onnx/backend/test/case/node/__init__.py

```diff
@@ -150,6 +160,31 @@ def expect(node: onnx.NodeProto,
         inputs=inputs_vi,
         outputs=outputs_vi)
     kwargs[str('producer_name')] = 'backend-test'
+
+    if 'opset_imports' not in kwargs:
+        # To make sure the model will be produced with the same opset_version after opset changes
+        # By default, it uses since_version as opset_version for produced models
+        if _TargetOpsetVersion is None:
+            produce_opset_version = onnx.defs.get_schema(node.op_type, node.domain).since_version
+        # If given targeted opset version
+        # it will generate test data for the latest opset vesion that supports before targeted opset version
+        else:
+            produce_opset_version = onnx.defs.get_schema(node.op_type, int(_TargetOpsetVersion), node.domain).since_version
+        kwargs[str('opset_imports')] = [onnx.helper.make_operatorsetid(node.domain, produce_opset_version)]
+        kwargs[str('ir_version')] = _OpsetVersionToIRVersionMap[produce_opset_version]
+    else:
+        # Also, for the same reason above, convert given opset.version (from opset_imports) to max_inclusive_version
+        max_inclusive_versions = []
+        onnx_opset_version = None
+        for opset in kwargs[str('opset_imports')]:
+            produce_opset_version = onnx.defs.get_schema(node.op_type, opset.version, opset.domain).since_version
+            max_inclusive_versions.append(onnx.helper.make_operatorsetid(opset.domain, produce_opset_version))
+            if opset.domain == '' or opset.domain == 'ai.onnx':
+                onnx_opset_version = produce_opset_version
+        kwargs[str('opset_imports')] = max_inclusive_versions
+        if onnx_opset_version is not None:
+            kwargs[str('ir_version')] = _OpsetVersionToIRVersionMap[onnx_opset_version]
```

Comment:
By default helper.make_model will produce a model with the latest IR_VERSION, which means giving an older opset_version make_model will create a model with older opset_version but with the latest ir_version. This looks problematic to me. Therefore, I use the version map here to get the proper ir_version according to the opset_version.

However, it even brings more updates than before. The reason is typically when a contributor introduced a new node test data, at that moment the opset_version has been already bumped, but the ir_version has not been bumped before the release. Thus most cases those node test data were using opset_version with (corresponding ir_version -1). This PR will make them use corresponding ir_version instead, which makes more sense to me.